HandEyeCalibration/Transform/Transform.cpp

```diff
@@ -0,0 +1,330 @@
+        }
+
+        Eigen::MatrixXd eigenMat(cvMat.rows, cvMat.cols);
+
+        for(int i = 0; i < cvMat.rows; i++)
+        {
+            for(int j = 0; j < cvMat.cols; j++)
+            {
+                eigenMat(i, j) = cvMat.at<double>(i, j);
+            }
+        }
+
+        return eigenMat;
+    }
+
+    Eigen::Vector4d reorderAxisAngle(const Eigen::AngleAxisd &axisAngle)
+    {
+        Eigen::Vector4d reorderedAxisAngle;
+
+        for(int i = 0; i < 3; i++)
+            reorderedAxisAngle[i] = axisAngle.axis()[i];
+        reorderedAxisAngle[3] = axisAngle.angle();
+
+        return reorderedAxisAngle;
+    }
+
+    /** Converts Rotation Matrix to RollPitchYaw.
+     * The convention we use here is that Roll is a rotation about x-axis,
+     * Pitch is a rotation about y-axis and Yaw is a rotation about z-axis.
+     * @param rotationMatrix Rotation Matrix to convert to RPY
+     * @param rotation Enum to indicate which convention to follow for this conversion
+     * @return rollPitchYaw array of angles, in radians
+     */
+    Eigen::Array3d rotmat2rpy(const Eigen::Matrix3d rotationMatrix,
+                              const TransformUtils::Rotation rotation = ZYX_INTRINSIC)
+    {
+        Eigen::Array3d rollPitchYaw;
+        switch(rotation)
+        {
+            case TransformUtils::Rotation::XYZ_INTRINSIC: rollPitchYaw = rotationMatrix.eulerAngles(0, 1, 2); break;
+            case TransformUtils::Rotation::XYZ_EXTRINSIC: rollPitchYaw = rotationMatrix.eulerAngles(2, 1, 0); break;
+            case TransformUtils::Rotation::ZYX_INTRINSIC: rollPitchYaw = rotationMatrix.eulerAngles(2, 1, 0); break;
+            case TransformUtils::Rotation::ZYX_EXTRINSIC: rollPitchYaw = rotationMatrix.eulerAngles(0, 1, 2); break;
+            default: throw "Invalid rotation";
+        }
+        // The returned array should be ordered by Ox, Oy and Oz. Where Oa is the rotation about axis a.
+        switch(rotation)
+        {
+            case TransformUtils::Rotation::XYZ_EXTRINSIC: rollPitchYaw.reverseInPlace(); break;
+            case TransformUtils::Rotation::ZYX_INTRINSIC: rollPitchYaw.reverseInPlace(); break;
+        }
+
+        return rollPitchYaw;
+    }
+
+    /** Converts RollPitchYaw to Rotation Matrix.
+     * This function takes an array of roll, pitch and yaw angles, and a convention, as input parameters.
+     * For rollPitchYaw we define that \a roll is a rotation about x-axis,
+     * \a pitch is a rotation about y-axis and \a yaw is a rotation about z-axis.
+     * @param rollPitchYaw Roll, Pitch and Yaw angles in radians. Order: Ox, Oy, Oz
+     * @param rotation Enum to indicate which convention to follow for this conversion
+     * @return rotationMatrix
+     */
+    Eigen::Matrix3d rpy2rotmat(const Eigen::Array3d rollPitchYaw,
+                               const TransformUtils::Rotation rotation = ZYX_INTRINSIC)
+    {
+        Eigen::Matrix3d rotationMatrix = Eigen::Matrix3d::Identity(3, 3);
```

Comment:
Do you need this variable? Could you return directly from the `case`s instead? Then I don't have to wonder what you're going to use the identity matrix for.

Reply:
Does the compiler understand that it won't ever get through the switch statement, and thus won't require a return in the end? I wouldn't know what to return, as NULL isn't accepted.

Reply:
That should work fine.